src/azul/service/storage_service.py

```diff
@@ -329,7 +329,7 @@ def _handle_overwrite(self,
                           object_key: str
                           ):
         error = exception.response['Error']
-        code, condition = error['Code'], error['Condition']
+        code, condition = error['Code'], error.get('Condition')
```

Comment:
PL, commit title misleading

Reply:
Added comment to explain when the KeyError occurs. Agreed at PL that this is sufficient.